client/js/components/GameView/MoveChoiceOverlay.vue

```diff
@@ -70,181 +70,173 @@ export default {
     },
   },
   computed: {
-    /**
-     * Returns list of objects representing the available moves,
-     * based on the selected card
-     */
-    moveChoices() {
-      if (!this.selectedCard) return [];
-      let res = [];
-      let cardRank;
-      if (this.resolvingSeven) {
-        if (!this.cardSelectedFromDeck) return [];
-        cardRank = this.cardSelectedFromDeck.rank;
-      } else {
-        if (!this.selectedCard) return [];
-        cardRank = this.selectedCard.rank;
-      }
-      // Re-usable move-objects
-      const allMovesAreDisabled = !this.isPlayersTurn || this.frozenId === this.selectedCard.id;
-      const allMovesDisabledExplanation = !this.isPlayersTurn
-        ? "It's not your turn"
-        : 'This card is frozen';
-
-      let disabledText;
-      if (allMovesAreDisabled) {
-        disabledText = allMovesDisabledExplanation;
+    // Determines if any moves are available
+    allMovesAreDisabled() {
+      return !this.isPlayersTurn || this.frozenId === this.selectedCard.id;
+    },
+    // Determines which disabled text to display
+    disabledText() {
+      if (this.allMovesAreDisabled) {
+        return !this.isPlayersTurn ? "It's not your turn" : 'This card is frozen';
       }
-
-      // Points
-      const pointsDescription = `Gain ${cardRank} point` + (cardRank === 1 ? '' : 's');
-      const pointsMove = {
+      return '';
+    },
+    pointsMove() {
+      if (!this.selectedCard) return [];
```

Comment:
Because this computed prop returns an object, the 'empty' case should be an empty object, or null, rather than an array.
```suggestion
      if (!this.selectedCard) return null;
```

Reply:
Should the `if (!this.selectedCard) return null;` check be added to each computed prop? I remember you saying the check should be required. I added it to a few as you can see, but im not sure I fully though through which ones to add/not add it to.

And, if we are adding to all of them, is there a better way we can do that instead of having it written in each one?

Reply:
I guess specifically, `allMovesAreDisabled()` and `disabledText()`

Reply:
Good point. This is a matter of type-safety. Essentially any time we reference a property on an object, we should first be sure that the object exists. I wrote earlier that we could assume there was always a selected card, and here I'm kindof waffling on that point. 

We either need to handle that possibility in every computed prop that references `this.selectedCard`, or we need to make the `selectedCard` prop required, and then update the `GameOverlays` so that this component is `v-if`'d to only appear when a card is selected. I think the second approach is actually cleaner. We can completely assume inside this component that `this.selectedCard` is defined and is a card, and therefore remove _all_ these null checks. The only thing is that we would still need all the computed props to return a value in all circumstances, so we'd just need to ensure there are no functions that only return inside a conditional like

```
someComputedProp() {
  if (something) {
    return whatever;
  }
}
```

That should instead be
```
someComputedProp() {
  if (something) {
    return whatever;
  }
  return whateverMakesSense;
}
```